### PR TITLE
Adding support for decimals

### PR DESCRIPTION
**Pull request by nvelozsavino**

### CODE REVIEW

**Comment by himelbrand** [comment]
thank you for the addition,
could you please add the new props to the README ?

File: NumericInput/NumericInput.js [hunk 1]
```diff
@@ -25,10 +25,20 @@ export default class NumericInput extends Component {
             this.setState({
                 value: props.initValue,
                 lastValid: props.initValue,
-                stringValue: props.initValue.toString()
+                stringValue: this.stringify(props.initValue)
             });
         }
     }
+    
+        componentDidUpdate(prevProps, prevState) {
```
Comment (himelbrand) [inline NumericInput/NumericInput.js]: Please fix indentation of componentDidUpdate.

File: NumericInput/NumericInput.js [hunk 2]
```diff
@@ -37,16 +47,25 @@ export default class NumericInput extends Component {
...
-            this.setState({ value, stringValue: value.toString() })
+            this.setState({ value, stringValue: this.stringify(value) })
 
         }
         if (value !== this.props.value)
             this.props.onChange && this.props.onChange(Number(value))
     }
+    stringify = (value) => {
+        if (this.props.decimals === null || this.props.decimals === undefined){
+            return value.toString();
```
Comment (himelbrand) [inline NumericInput/NumericInput.js]: Please remove semi-colon

File: NumericInput/NumericInput.js [hunk 3]
```diff
@@ -158,7 +177,7 @@ export default class NumericInput extends Component {
         const iconStyle = [style.icon, this.props.iconStyle]
         const totalWidth = this.props.totalWidth
         const totalHeight = this.props.totalHeight ? this.props.totalHeight : (totalWidth * 0.4)
-        const inputWidth = this.props.type === 'up-down' ? (totalWidth * 0.6) : (totalWidth * 0.4)
+        const inputWidth = this.props.inputWidth? this.props.inputWidth: this.props.type === 'up-down' ? (totalWidth * 0.6) : (totalWidth * 0.4)
```
Comment (himelbrand) [inline NumericInput/NumericInput.js]: Please add space between this.props.inputWidth and ?  and also between this.props.inputWidth and :

File: NumericInput/NumericInput.js (hunk 1 above)
Comment (himelbrand) [reply NumericInput/NumericInput.js]: adding to review

**Comment by himelbrand** [comment]
@nvelozsavino until you make the changes I will not approve this PR.
if you don't have time, let me know and I'll make the changes myself.
thanks

**Comment by nvelozsavino** [comment]
I haven't had any time for it, sorry. Also the widget is still buggy when editing. However I don't think I can work on it any time soon.

**Pull request closed without merging**